storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/RemoteLogMetadataCache.java

```diff
@@ -302,22 +307,26 @@ public void addCopyInProgressSegment(RemoteLogSegmentMetadata remoteLogSegmentMe
 
         RemoteLogSegmentId remoteLogSegmentId = remoteLogSegmentMetadata.remoteLogSegmentId();
         RemoteLogSegmentMetadata existingMetadata = idToSegmentMetadata.get(remoteLogSegmentId);
-        checkStateTransition(existingMetadata != null ? existingMetadata.state() : null,
-                remoteLogSegmentMetadata.state());
-
+        boolean isValid = checkStateTransition(existingMetadata != null ? existingMetadata.state() : null,
+                remoteLogSegmentMetadata.state(), remoteLogSegmentMetadata.remoteLogSegmentId());
+        if (!isValid) {
+            return;
+        }
         for (Integer epoch : remoteLogSegmentMetadata.segmentLeaderEpochs().keySet()) {
             leaderEpochEntries.computeIfAbsent(epoch, leaderEpoch -> new RemoteLogLeaderEpochState())
                     .handleSegmentWithCopySegmentStartedState(remoteLogSegmentId);
         }
-
         idToSegmentMetadata.put(remoteLogSegmentId, remoteLogSegmentMetadata);
     }
 
-    private void checkStateTransition(RemoteLogSegmentState existingState, RemoteLogSegmentState targetState) {
-        if (!RemoteLogSegmentState.isValidTransition(existingState, targetState)) {
-            throw new IllegalStateException(
-                    "Current state: " + existingState + " can not be transitioned to target state: " + targetState);
+    private boolean checkStateTransition(RemoteLogSegmentState existingState,
+                                         RemoteLogSegmentState targetState,
+                                         RemoteLogSegmentId segmentId) {
+        boolean isValid = RemoteLogSegmentState.isValidTransition(existingState, targetState);
+        if (!isValid) {
+            log.error("Current state: {} can not be transitioned to target state: {}, segmentId: {}. Dropping the event",
```

Comment:
Logging the error instead of throwing the exception as it will stop the internal consumer which consumes from the remote log metadata topic.

Reply:
`enable.idempotence ` is set to true by default from 3.0.1 actually.